DependencyInjection/Compiler/FallbackServiceRepositoryCompilePass.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Common\Persistence\Mapping\ClassMetadataFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class FallbackServiceRepositoryCompilePass implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     *
+     * @throws \Exception
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach (array_keys($container->getDefinitions()) as $id) {
+            if (!preg_match('/^doctrine\.orm\.[^.]+_entity_manager$/', $id)) {
+                continue;
+            }
+
+            $this->processMetadata(
+                $container,
+                $id,
+                $container->get($id)->getMetadataFactory()
```

Comment:
Instantiating a service in a compiler pass is a total no-go. There is exactly **no** guarantee that this will work, as the container is not fully configured yet. And even if it works at some point in time in your project, any change to any service definition could break it. So doing that in an open-source bundle which has to work in any project is impossible.